llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/base.py

```diff
@@ -0,0 +1,204 @@
+import aiohttp
+import requests
+from typing import Any, List, Optional
+from tenacity import Retrying, RetryError, stop_after_attempt, wait_fixed
+
+from llama_index.core.base.embeddings.base import (
+    DEFAULT_EMBED_BATCH_SIZE,
+    BaseEmbedding,
+)
+
+from llama_index.core.bridge.pydantic import Field
+from llama_index.core.callbacks import CallbackManager
+from llama_index.core.base.llms.generic_utils import get_from_param_or_env
+from llama_index.embeddings.yandexgpt.util import YException
+
+DEFAULT_YANDEXGPT_API_BASE = (
+    "https://llm.api.cloud.yandex.net/foundationModels/v1/textEmbedding"
+)
+
+
+class YandexGPTEmbedding(BaseEmbedding):
+    """
+    A class representation for generating embeddings using the Yandex Cloud API.
+
+    Args:
+      api_key (Optional[str]): An API key for Yandex Cloud.
+      model_name (str): The name of the model to be used for generating embeddings.
+                         The class ensures that this model is supported. Defaults to "general:embedding".
+      embed_batch_size (int): The batch size for embedding. Defaults to DEFAULT_EMBED_BATCH_SIZE.
+      callback_manager (Optional[CallbackManager]): Callback manager for hooks.
+
+    Example:
+        . code-block:: python
+
+            from llama_index.embeddings.yandexgpt import YandexGPTEmbedding
+
+            embeddings = YandexGPTEmbedding(
+                api_key="your-api-key",
+                folder_id="your-folder-id",
+            )
+    """
+
+    api_key: str = Field(description="The YandexGPT API key.")
+    folder_id: str = Field(description="The folder id for YandexGPT API.")
+    retries: int = 6
+    sleep_interval: float = 0.1
+
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        folder_id: Optional[str] = None,
+        model_name: str = "general:embedding",
+        embed_batch_size: int = DEFAULT_EMBED_BATCH_SIZE,
+        callback_manager: Optional[CallbackManager] = None,
+        **kwargs: Any,
+    ) -> None:
+        if not api_key:
+            raise ValueError(
+                "You must provide an API key or IAM token to use YandexGPT. "
+                "You can either pass it in as an argument or set it `YANDEXGPT_API_KEY`."
+            )
+        if not folder_id:
+            raise ValueError(
+                "You must provide catalog_id to use YandexGPT. "
+                "You can either pass it in as an argument or set it `YANDEXGPT_CATALOG_ID`."
+            )
+
+        api_key = get_from_param_or_env("api_key", api_key, "YANDEXGPT_KEY")
+        folder_id = get_from_param_or_env(
+            "folder_id", folder_id, "YANDEXGPT_CATALOG_ID"
+        )
+
+        super().__init__(
+            model_name=model_name,
+            api_key=api_key,
+            folder_id=folder_id,
+            embed_batch_size=embed_batch_size,
+            callback_manager=callback_manager,
+            **kwargs,
+        )
+
+    def _getModelUri(self, is_document=False) -> str:
```

Comment:
Can we please type the functions. `def _getModelUri(self, is_document: bool = False) -> str:`